src/openforms/contrib/zgw/clients/catalogi.py

```diff
@@ -227,6 +234,37 @@ def find_case_types(
 
         return all_versions
 
+    def find_case_types_products(
+        self,
+        *,
+        catalogus: str,
+        identification: str,
+        valid_on: date | None = None,
+    ) -> list[Product]:
+        products: list[Product] = []
+        case_versions = self.find_case_types(
+            catalogus=catalogus,
+            identification=identification,
+            valid_on=valid_on,
+        )
+
+        if (
+            case_versions
+            and len(case_versions) > 0
+            and "productenOfDiensten" in case_versions[0]
+        ):
+            for product_uri in case_versions[0]["productenOfDiensten"]:
+                description = ""
+
+                response = requests.get(
+                    product_uri, headers={"Content-Type": "application/json"}
+                )
+                if response.ok:
+                    description = response.json().get("description")
+                products.append(Product(uri=product_uri, description=description))
+
+        return products
```

Comment:
I'm trying to figure out if this method should exist on the Catalogi client or not. Technically, the product/service URLs are not part of the Catalogi API, so it should be a different client because it's a different domain. However, there is no formal specification yet, so we don't really have a good place for the code.

So, I'd propose this:

```py
# in file src/openforms/contrib/zgw/products_and_services.py
"""
Support resolving products and services referenced by case types.

.. warning:: Products and services integration is experimental. There is no standard for it yet.
"""

def resolve_case_type_products(case_type_versions: list[CaseType]) -> list[Product]:
    all_urls: set[str] = set(sum(version["productenOfDiensten"] for version in case_type_versions, []))
    ...  # look up content of each URL
    return products
```

and then in `src/openforms/contrib/zgw/service.py` expose this helper:

```py
from .products_and_services import resolve_case_type_products
```

You can then combine these things at the places where you need them:

```py
case_type_versions = client.find_case_types(...)
products = resolve_case_type_products(case_type_versions) if case_type_versions else []
```

And this way, we keep the clients/domains pure.

Reply:
Yeah this thought also crossed my mind. I decided this would be an okay place, seeing how it uses the catalogi service to fetch the case_type data, but fetching the products was a bit weird..

I've moved this logic like you suggested, and removed the `find_case_types_products` function. The view now first fetches the case type versions, and then gets the related products with extra information